lib/improver/utilities/solar.py

```diff
@@ -0,0 +1,259 @@
+
+    Args:
+        longitudes (float or numpy.array):
+            A single Longitude or array of Longitudes
+            longitudes needs to be between 180.0 and -180.0
+        day_of_year (int):
+            Day of the year 0 to 365, 0 = 1st January
+        utc_hour (float):
+            Hour of the day in UTC
+
+    Returns:
+        solar_hour_angle (numpy.array)
+            Hour angle in degrees East-West
+    """
+    thetao = 2*np.pi*day_of_year/365.0
+    eqt = (0.000075 + 0.001868 * np.cos(thetao) -
+           0.032077 * np.sin(thetao) - 0.014615 * np.cos(2*thetao) -
+           0.040849 * np.sin(2*thetao))
+
+    # Longitudinal Correction from the Grenwich Meridian
+    lon_correction = 24.0*longitudes/360.0
+    # Solar time (hours):
+    solar_time = utc_hour + lon_correction + eqt*12/np.pi
+    # Hour angle (degrees):
+    solar_hour_angle = (solar_time - 12.0) * 15.0
+
+    return solar_hour_angle
+
+
+def solar_elevation(latitudes, longitudes, day_of_year, utc_hour):
+    """
+    Calculate the Solar elevation.
+
+    Args:
+        latitudes (float or numpy.array):
+            A single Latitude or array of Latitudes
+            latitudes needs to be between -90.0 and 90.0
+        longitudes (float or numpy.array):
+            A single Longitude or array of Longitudes
+            longitudes needs to be between 180.0 and -180.0
+        day_of_year (int):
+            Day of the year 0 to 365, 0 = 1st January
+        utc_hour (float):
+            Hour of the day in UTC in hours
+
+    Returns:
+        solar_elevation (numpy.array):
+            Solar elevation in degrees
+    """
+    declination = solar_declination(day_of_year)
+    decl = np.radians(declination)
+    hour_angle = solar_hour_angle(longitudes, day_of_year, utc_hour)
+    rad_hours = np.radians(hour_angle)
+    lats = np.radians(latitudes)
+    # Calculate solar position:
+
+    solar_elevation = (np.arcsin(np.sin(decl) * np.sin(lats) +
+                                 np.cos(decl) * np.cos(lats) *
+                                 np.cos(rad_hours)))
+
+    solar_elevation = np.degrees(solar_elevation)
+
+    return solar_elevation
+
+
+def daynight_terminator(longitudes, day_of_year, utc_hour):
+    """
+    Calculate the Latitude values of the daynight terminator
+
+    Args:
+        longitudes (numpy.array):
+            Array of longitudes
+        day_of_year (int):
+            Day of the year
+        utc_hour (float):
+            Hour of the day in UTC
+
+    Returns:
+        latitudes (numpy.array):
+            latitudes of the daynight terminator
+    """
+    declination = solar_declination(day_of_year)
+    decl = np.radians(declination)
+    hour_angle = solar_hour_angle(longitudes, day_of_year, utc_hour)
+    rad_hour = np.radians(hour_angle)
+    lats = np.arctan(-np.cos(rad_hour)/np.tan(decl))
+    lats = np.degrees(lats)
+    return lats
+
+
+class DayNightMask(object):
+    """
+    Generate a daynight mask for the provided cube
```

Comment:
The class itself doesn't do anything, its functions do things. How about "Plugin class to generate..."

Reply:
Changed